src/app/view/device/add-devices/add-devices.component.ts

```diff
@@ -405,33 +573,59 @@ export class AddDevicesComponent {
           }
         },
         error: (err) => {
-          //Error callback
+          // Error callback
           console.error('error caught in component', err.error.message);
           if (err.error.statusCode === 403) {
             this.toastrService.error(
               "You don't have the permissions to add a device.",
               'Access Denied',
             );
           } else {
+            this.submitButtonText = 'Submit';
+            this.isSubmitting = false;
             this.toastrService.error(
-              'some error occurred due to ' + err.error.message,
+              'Some error occurred due to ' + err.error.message,
               'Device!' + element.externalId,
             );
           }
         },
       });
     });
   }
+  shortenFileName(fileName: string, maxLength: number = 20): string {
+    if (!fileName || fileName.length <= maxLength) {
```

Comment:
This one should also be a helper function. This is not particular to device file upload